spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/authentication/OAuth2AuthenticationProcessingFilter.java

```diff
@@ -176,6 +178,14 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
 		chain.doFilter(request, response);
 	}
 
+	protected void onSuccessfulAuthentication(HttpServletRequest request, HttpServletResponse response,
```

Comment:
The preferred approach would be to add `AuthenticationSuccessHandler` and `AuthenticationFailureHandler` collaborators. See `AbstractAuthenticationProcessingFilter`